metriken-exposition/src/snapshot.rs

```diff
@@ -0,0 +1,53 @@
+use crate::*;
+
+/// Contains a snapshot of metric readings.
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Snapshot {
```

Comment:
Also this should probably be implementing `Debug` and `Clone`.

Reply:
histogram snapshot doesn't implement Debug. I'm probably going to go make some changes in there anyway. Putting a TODO for debug impl.